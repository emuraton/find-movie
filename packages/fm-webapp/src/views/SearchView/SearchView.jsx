import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import request from 'superagent'
import { SearchBar, LayoutGrid } from 'fm-components'
import env from '../../../env.json'
import { setMovies } from '../../actions/moviesAction'

class SearchView extends Component {
  constructor(props) {
    super(props)
  }

  callSearch = (query) => {
    const { dispatch } = this.props
    request
      .get(`${env.API_URL}${env.ENDPOINT_SEARCH_MOVIE}`)
      .query({ api_key: env.API_KEY, query, page: 1 })
      .then(({ body }) => {
        dispatch(setMovies(body.results))
      })
      .catch((e) => {
        // TODO Improve error management with a message error in the UI
        dispatch(setMovies())
      })
  }

  render() {
    const { movies } = this.props
    return (
      <section className="SearchView">
        <SearchBar onSubmit={this.callSearch} />
        { movies ? <LayoutGrid data={movies} /> : null }
      </section>
    )
  }
}

SearchView.PropTypes = {
  movies: PropTypes.object,
}

export default connect(
  state => ({ movies: state.movies.list })
)(SearchView)
