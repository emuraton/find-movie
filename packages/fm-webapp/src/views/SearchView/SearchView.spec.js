import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

jest.mock('fm-components', () => ({
    SearchBar: 'SearchBarMocked',
    LayoutGrid: 'LayoutGridMocked'
}))

import SearchView from './SearchView'

describe('<SearchView />', () => {
  describe('@renders', () => {
    it('in default state', () => {
      const wrapper = shallow(<SearchView.WrappedComponent />)
      expect(toJson(wrapper))
        .toMatchSnapshot()
    })

    it('in with movies state', () => {
      const props = {
        movies: {
          1: {},
          2: {}
        }
      }
      const wrapper = shallow(<SearchView.WrappedComponent {...props}/>)
      expect(toJson(wrapper))
        .toMatchSnapshot()
    })
  })
})
