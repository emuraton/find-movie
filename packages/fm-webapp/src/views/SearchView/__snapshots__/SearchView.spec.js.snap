// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchView /> @renders in default state 1`] = `
<section
  className="SearchView"
>
  <SearchBarMocked
    onSubmit={[Function]}
  />
</section>
`;

exports[`<SearchView /> @renders in with movies state 1`] = `
<section
  className="SearchView"
>
  <SearchBarMocked
    onSubmit={[Function]}
  />
  <LayoutGridMocked
    data={
      Object {
        "1": Object {},
        "2": Object {},
      }
    }
  />
</section>
`;
